internal/app/secretless/providers/conjur/provider.go

```diff
@@ -271,45 +231,40 @@ func (p *Provider) refreshAccessToken() error {
 	// Authenticate in a loop with retries on failure with exponential backoff
 	err = backoff.Retry(func() error {
 		for {
-
 			// Lock the authenticatorMutex
 			p.AuthenticationMutex.Lock()
 
 			log.Printf("Info: Conjur provider is authenticating as %s ...", p.Authenticator.Config.Username)
 			resp, err := p.Authenticator.Authenticate()
-
-			if err == nil {
-				log.Printf("Info: Conjur provider received a valid authentication response")
-				err = p.Authenticator.ParseAuthenticationResponse(resp)
-			}
-
 			if err != nil {
+				p.AuthenticationMutex.Unlock()
 				log.Printf("Info: Conjur provider received an error on authenticate: %s", err.Error())
+				return err
+			}
 
-				if autherr, ok := err.(*authenticator.Error); ok {
-					if autherr.CertExpired() {
-						log.Printf("Info: Conjur certificate expired; Conjur provider is re-logging in.")
-
-						if err = p.Authenticator.Login(); err != nil {
-							return err
-						}
+			log.Printf("Info: Conjur provider received a valid authentication response")
 
-						// if the cert expired and login worked then continue
-						continue
-					}
-				} else {
-					return fmt.Errorf("Error: Conjur provider unable to authenticate: %s", err.Error())
-				}
+			err = p.Authenticator.ParseAuthenticationResponse(resp)
+			if err != nil {
+				p.AuthenticationMutex.Unlock()
+				log.Printf("Failure parsing response: %s", err.Error())
```

Comment:
Maybe the error should be prefixed by "Conjur provider" to be clear which internal component is erroring